src/uct/ib/mlx5/rc/rc_mlx5_devx.c

```diff
@@ -270,12 +274,35 @@ uct_rc_mlx5_devx_init_rx_common(uct_rc_mlx5_iface_common_t *iface,
                           log_num_of_strides & 0xF);
         UCT_IB_MLX5DV_SET(wq, wq, log_wqe_stride_size,
                           (ucs_ilog2(iface->super.super.config.seg_size) - 6));
+    } else if (uct_rc_mlx5_iface_is_srq_msg_based(iface)) {
+        ucs_assertv(iface->super.super.config.stride_size <=
```

Comment:
both `stride_size` and `seg_size` are config attrs, so maybe worth validate them in run time and print error? also, what if `seg_size` not aligned to `stride_size`?